structure_tests/types/v2/structure.go

```diff
@@ -141,6 +143,88 @@ func (st *StructureTest) RunFileContentTests(t *testing.T) int {
 	return counter
 }
 
+func (st *StructureTest) RunMetadataTests(t *testing.T) int {
+	t.Run(st.MetadataTest.LogName(), func(t *testing.T) {
+		driver, err := st.NewDriver()
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+		defer driver.Destroy()
+		config, err := driver.GetConfig(t)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+		for _, pair := range st.MetadataTest.Env {
+			if config.Env[pair.Key] == "" {
+				t.Errorf("variable %s not found in image env", pair.Key)
+			} else if config.Env[pair.Key] != pair.Value {
+				t.Errorf("env var %s value does not match expected value: %s", pair.Key, pair.Value)
+			}
+		}
+
+		if len(st.MetadataTest.Cmd) > 0 {
+			// legitimate cmd provided by user in config
+			if st.MetadataTest.Cmd[0] != CMD_PLACEHOLDER {
+				// mismatched lengths auto fails
+				if len(st.MetadataTest.Cmd) != len(config.Cmd) {
+					t.Errorf("Image Cmd %v does not match expected Cmd: %v", st.MetadataTest.Cmd, config.Cmd)
+				}
+				for i := range st.MetadataTest.Cmd {
+					if st.MetadataTest.Cmd[i] != config.Cmd[i] {
+						t.Errorf("Image config Cmd does not match expected value: %s", st.MetadataTest.Cmd)
+					}
+				}
+			}
+		} else if len(config.Cmd) > 0 {
+			// explicit empty cmd was provided by user but image has one: fail
+			t.Errorf("Image Cmd expected to be empty but is %v instead", config.Cmd)
+		}
+
+		if len(st.MetadataTest.Entrypoint) > 0 {
+			// legitimate entrypoint provided by user in config
+			if st.MetadataTest.Entrypoint[0] != CMD_PLACEHOLDER {
+				// mismatched lengths auto fails
+				if len(st.MetadataTest.Entrypoint) != len(config.Entrypoint) {
+					t.Errorf("Image entrypoint %v does not match expected Cmd: %v", st.MetadataTest.Entrypoint, config.Entrypoint)
+				}
+				for i := range st.MetadataTest.Entrypoint {
+					if st.MetadataTest.Entrypoint[i] != config.Entrypoint[i] {
+						t.Errorf("Image config entrypoint does not match expected value: %s", st.MetadataTest.Entrypoint)
+					}
+				}
+			}
+		} else if len(config.Entrypoint) > 0 {
+			// explicit empty entrypoint was provided by user but image has one: fail
+			t.Errorf("Image entrypoint expected to be empty but is %v instead", config.Entrypoint)
+		}
+
+		if st.MetadataTest.Workdir != "" && st.MetadataTest.Workdir != config.Workdir {
+			t.Errorf("Image Workdir %s does not match config Workdir: %s", st.MetadataTest.Workdir, config.Workdir)
+		}
+
+	PortLoop:
```

Comment:
Not a big fan of the label pattern, but I see why you used it here. Maybe break it out as a helper function?

Reply:
done